lib/global-styles.php

```diff
@@ -492,6 +492,16 @@ function gutenberg_experimental_global_styles_resolver_styles( $block_selector,
 	$css_rule         = '';
 	$css_declarations = '';
 
+	// Transformations for block-supports strings.
+	foreach ( $block_supports as $block_support ) {
+		// Convert camelCase to kebab-case and add to array.
+		// Don't check if the item already exists with in_array(), we will remove duplicate values outside the loop.
```

Comment:
I'll check it more on Monday, but my first though is why to skip the `if exists` check here in this loop, and then have to remove the duplicates. I can't see any performance gain. I checked that the `array_flip` below has been benchmarked and seem to be faster than `array_unique`, but it doesn't explain the skip of the `if` check.

Is there a reason I can't see here?

Reply:
The alternative would be this:
```php
foreach ( $block_supports as $block_support ) {
	$kebab_case = strtolower( preg_replace( '/([a-z0-9]|(?=[A-Z]))([A-Z])/', '$1-$2', $block_support ) );
	if ( ! in_array( $kebab_case, $block_supports ) ) {
		$block_supports[] = $kebab_case;
	}
}
```
The problem is that the `gutenberg_experimental_global_styles_resolver_styles()` function is called inside a `foreach()` loop so it runs multiple times (once for each block). If a site has 100 blocks and 5 `$block_supports` per-block-type, that adds up to 500 `in_array()` checks.
That's why I chose to skip the checks and do the `array_flip( array_flip() )` instead, it will save a few hundred function calls. On their own the checks may not be expensive, but we have nested `foreach()` loops.
Adding the items to the array without checking for duplicates and then remove any duplicates seemed like the better option in this case.

If using the more "traditional" approach of adding `in_array()` checks is deemed preferable then we can use that...